clients/ui/frontend/src/app/pages/modelCatalog/utils/hardwareConfigurationFilterUtils.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  CatalogPerformanceMetricsArtifact,
+  ModelCatalogFilterStates,
+  PerformanceMetricsCustomProperties,
+} from '~/app/modelCatalogTypes';
+import { getDoubleValue, getStringValue } from '~/app/utils';
+import {
+  ModelCatalogStringFilterKey,
+  ModelCatalogNumberFilterKey,
+  LatencyMetricKey,
+} from '~/concepts/modelCatalog/const';
+import { getTotalRps } from './performanceMetricsUtils';
+
+// Type for storing complex latency filter configuration
+export type LatencyFilterConfig = {
+  metric: 'E2E' | 'TTFT' | 'TPS' | 'ITL';
+  percentile: 'Mean' | 'P90' | 'P95' | 'P99';
+  value: number;
+};
+
+/**
+ * Maps metric and percentile combination to the corresponding artifact field
+ */
+const getLatencyFieldName = (
+  metric: string,
+  percentile: string,
+): keyof PerformanceMetricsCustomProperties => {
+  const metricPrefix = metric.toLowerCase();
+  const percentileSuffix = percentile === 'Mean' ? '_mean' : `_${percentile.toLowerCase()}`;
+  const fieldName = `${metricPrefix}${percentileSuffix}`;
+
+  // Validate that the field exists in PerformanceMetricsCustomProperties
+  // Get latency fields from LatencyMetricKey enum
+  const validFields = Object.values(LatencyMetricKey);
+
+  return validFields.some((field) => field === fieldName)
+    ? // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+      (fieldName as keyof PerformanceMetricsCustomProperties)
```

Comment:
Ok I see - yeah this may be harder to avoid, will revisit this. some combination of type guards and more narrow param types should help.